src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -168,12 +174,130 @@
+
+(defn calculate-input-text [{:keys [full-text selection-start selection-end]} content]
+  (let [head (subs full-text 0 selection-start)
+        tail (subs full-text selection-end)]
+    (str head content tail)))
+
+(def first-level-menus {:cut (fn [{:keys [content] :as params}]
+                               (let [new-text (calculate-input-text params "")]
+                                 (react/copy-to-clipboard content)
+                                 (update-input-text params new-text)))
+
+                        :copy-to-clipboard (fn [{:keys [content]}]
+                                             (react/copy-to-clipboard content))
+
+                        :paste (fn [params]
+                                 (let [callback (fn [paste-content]
+                                                  (let [content (string/trim paste-content)
+                                                        new-text (calculate-input-text params content)]
+                                                    (update-input-text params new-text)))]
+                                   (react/get-from-clipboard callback)))
+
+                        :biu (fn [{:keys [first-level text-input-handle menu-items selection-start selection-end]}]
+                               (let [show-menu #(.startActionMode (selectable-text-input-manager) text-input-handle)]
+                                 (reset! first-level false)
+                                 (reset! menu-items second-level-menu-items)
+                                 ;to avoid something discusting like this https://lightrun.com/answers/facebook-react-native-textinput-controlled-selection-broken-on-both-ios-and-android
+                                 ;use native invoke instead! do not use setNativeProps! e.g. (.setNativeProps ^js text-input (clj->js {:selection {:start selection-start :end selection-end}}))
+                                 (.setSelection (selectable-text-input-manager) text-input-handle selection-start selection-end)
+                                 ;set a delay to wait RNSelectableTextInputViewManager#_menuItems get updated, need figure out a better way
+                                 (js/setTimeout show-menu 200)))})
+
+(def first-level-menu-items (map i18n/label (keys first-level-menus)))
+
+(defn reset-to-first-level-menu [first-level menu-items]
+  (reset! first-level true)
+  (reset! menu-items first-level-menu-items))
+
+(defn append-markdown-char [{:keys [first-level menu-items content] :as params} wrap-chars]
+  (let [content (str wrap-chars content wrap-chars)
+        new-text (calculate-input-text params content)]
+    (update-input-text params new-text)
+    (reset-to-first-level-menu first-level menu-items)))
+
+(def second-level-menus {:bold #(append-markdown-char % "**")
+
+                         :italic #(append-markdown-char % "*")
+
+                         :strikethrough #(append-markdown-char % "~~")})
```

Comment:
Kindly correct me if am wrong ..
I see a `:biu` keyword in the code which I assume it stands for bold/italic/underline
But I can't seem to find an underlining option anywhere ..

Reply:
it comes from the design